src/lib/parser.rs

```diff
@@ -108,8 +108,47 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
-    let letter = self.contents.get(self.index)?;
+    let letter = *self.contents.get(self.index)? as char;
+
+    let fs = '/' as u8;
+    let nl = '\n' as u8;
+    let ast = '*' as u8;
+
     self.index += 1;
+
+    // check for forward slash
+    if letter == &fs {
+      // check for next forward slash
+      let next = self.contents.get(self.index)?;
+      if next == &fs {
+        // detected single line comment
+        // loop until newline (comments are not parsed)
+        loop {
+          let next = self.contents.get(self.index)?;
+          self.index += 1;
+          // check for newline (end of comment)
+          if next == &nl {
+            return self.next_char();
+          }
+        }
+      } else if letter == &ast {
+        // detected multi-line comment
+        // loop until closed (comments are not parsed)
+        loop {
+          let next = self.contents.get(self.index)?;
+          self.index += 1;
+          if next == &ast {
+            // * detected
+            let last = self.contents.get(self.index)?;
+            if last == &ast {
+              // */ detected
+              self.index += 1;
+              break
```

Comment:
I think you need to also return the next char here :)
```suggestion
             return self.next_char();
```